File: common/src/main/java/com/rabtman/common/base/mvp/IView.java
package com.rabtman.common.base.mvp;

public interface IView {

  /**
   * 显示加载
   */
  void showLoading();

  /**
   * 隐藏加载
   */
  void hideLoading();

  /**
   * 显示错误信息
   */
  void showError(String message);

  /**
   * 显示错误信息
   */
  void showError(int stringId);

  /**
   * 显示信息
   */
  void showMsg(String message);

  /**
   * 显示信息
   */
  void showMsg(int stringId);

  /**
   * 页面加载中
   */
  void showPageLoading();

  /**
   * 空白页面
   */
  void showPageEmpty();

  /**
   * 页面加载失败
   */
  void showPageError();

  /**
   * 展示页面内容
   */
  void showPageContent();
}

File: common/src/main/java/com/rabtman/common/base/mvp/IView.kt
package com.rabtman.common.base.mvp

interface IView {
    /**
     * 显示加载
     */
    fun showLoading()

    /**
     * 隐藏加载
     */
    fun hideLoading()

    /**
     * 显示错误信息
     */
    fun showError(message: String?)

    /**
     * 显示错误信息
     */
    fun showError(stringId: Int)

    /**
     * 显示信息
     */
    fun showMsg(message: String?)

    /**
     * 显示信息
     */
    fun showMsg(stringId: Int)

    /**
     * 页面加载中
     */
    fun showPageLoading()

    /**
     * 空白页面
     */
    fun showPageEmpty()

    /**
     * 页面加载失败
     */
    fun showPageError()

    /**
     * 展示页面内容
     */
    fun showPageContent()
}